tests/js/realm-tests.js

```diff
@@ -325,6 +332,16 @@ module.exports = BaseTest.extend({
                 realm.create('InvalidObject', {intCol: 1});
             });
         });
+
+        // Only the original constructor should be valid.
+        function WrongCustomObject() {}
+        WrongCustomObject.schema = CustomObject.schema;
```

Comment:
Kind of unrelated, but maybe we need to freeze the schema or constructor if a user uses it when opening a realm. Would be weird if a user changed the schema after opening a realm.